### ISSUE TITLE
question about algorithm 2&3

### ISSUE THREAD

**Issue by jacksonsc007** [Gsunshine/phantom_grad#1]
Hi, thanks for your excellent and impressive work. There might be silly question about the algorithm 2&3:

In my own calculation, it seems that lambda is not necessary when calculating the g_out, as shown below:
![image](https://github.com/Gsunshine/phantom_grad/assets/62741471/7cfd6c44-9c37-41a3-9c11-58e162825c68)

If I didn't make a mistake, the partial derivative of f with respect to z has already contained the lambda, as shown below. Take algorithm 2 for example, in the backward pass, h is treated as a constant vector which has nothing to do with z, due to the torch.no_grad() context.

![image](https://github.com/Gsunshine/phantom_grad/assets/62741471/759a2d13-36fd-4f78-98a2-274a23e121ed)

Please don't mind my poor math if I am wrong.

**Comment by Gsunshine** [comment]
Hi @jacksonsc007 ! You are right! It should be $\frac{\partial f}{\partial z}$ or $\lambda \frac{\partial F}{\partial z}$. This will not change the direction of the gradient but the magnitude.

Thank you for catching the bug! :)

**Comment by jacksonsc007** [comment]
@Gsunshine Thanks for your excellent work!

**Comment by Gsunshine** [comment]
thx! BTW, here is [torchdeq](https://github.com/locuslab/torchdeq) and [arxiv](https://arxiv.org/abs/2310.18605)! It could be helpful to carry out studies on DEQs.

Let me know if you have further questions!

**Comment by jacksonsc007** [comment]
Thanks for your suggestions.
